### PR TITLE
[enh] Avatar video support

### PR DESCRIPTION
**Pull request by ronronscelestes**
## What

- Adds video support for `Avatar`
- Allows `background` and `textColor` customization for `Initials`

## Test

- Use local video files and test `Avatar` component

### CODE REVIEW

**Review by gu-stav — CHANGES_REQUESTED** [review]
I think we need to discuss this approach. I'm not a big fan, because the behavior is very specific to the Strapi monorepo. Haven't thought about it much yet, but I think this is a good case to improve composability of the DS instead of passing it through a prop.

File: packages/strapi-design-system/src/helpers/throwPropError.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  return PropTypes.checkPropTypes({ [propName]: PropTypes[propType] }, props, 'prop', componentName);
+};
+
+/**
+ * @type {(otherProps: any{}) => (props: Record<string, unknown>, propName: string) => Error | undefined}
+ */
+
+export const throwPropErrorRequiredIf = (otherProps, propType) => (props, propName, componentName) => {
```
Comment (joshuaellis) [inline packages/strapi-design-system/src/helpers/throwPropError.js:19]:
```suggestion
export const throwPropErrorIfRequired = (otherProps, propType) => (props, propName, componentName) => {
```

Comment (joshuaellis) [reply packages/strapi-design-system/src/helpers/throwPropError.js:19]: found that a bit hard to read to begin with?

**Review by joshuaellis — CHANGES_REQUESTED** [review]
Small request, rest of the work is lovely to see ⭐

File: packages/strapi-design-system/src/helpers/throwPropError.js (hunk 1 above)
Comment (ronronscelestes) [reply packages/strapi-design-system/src/helpers/throwPropError.js:19]: works for me 😊

**Comment by gu-stav** [comment]
My main concern here is that we are overloading the `Avatar` component.

If we would say `Avatar`s can also be videos, I think that could work. The whole component doesn't make sense in the context of a list (because we are not rendering avatars, but assets).

Instead, I'd suggest we keep `Avatar` as limited as it is and create a new component - not sure what to call it, but it could be something about "assets" and "previews" because that's what it does in the end. An avatar is way more specific.

I think this could also be an interesting issue to discuss with design, because:

- this component is too small to render video player controls (so it would basically render a static image of the video)
- given that, how is the `preview` supposed to work? Shouldn't we do our best to preview a video player that can be interacted with? That would mean to take a different approach which doesn't have to be compatible with `Avatar`.
- I find that implementation very opinionated and specific to the CMS and I don't see how this would be useful e.g. in the context of the cloud

**Comment by ronronscelestes** [comment]
I'll close this PR for now as we would need to discuss and refine this idea of an asset preview component

**Comment by gu-stav** [comment]
@ronronscelestes Just curious: does that mean you agree? It's ok to disagree with me here :)

**Comment by ronronscelestes** [comment]
> @ronronscelestes Just curious: does that mean you agree? It's ok to disagree with me here :)

I agree it feels like extending this component just for 1 CMS current need that is not really related to what an avatar means 
I also like the idea to rethink the hover/preview for the video - and that would make more sense to do that in a Preview component

**Pull request closed without merging**